block-storage/src/main/scala/coop/rchain/blockstorage/BlockStore.scala

```diff
@@ -0,0 +1,44 @@
+package coop.rchain.blockstorage
+
+import cats.{Applicative, MonadError}
+import cats.effect.{Bracket, Sync}
+import cats.implicits._
+import com.google.protobuf.ByteString
+import coop.rchain.casper.protocol.BlockMessage
+import coop.rchain.metrics.Metrics
+
+import scala.language.higherKinds
+
+trait BlockStore[F[_]] {
+  import BlockStore.BlockHash
+
+  implicit def applicative: Applicative[F]
```

Comment:
The mtl `trait`-s I've seen so far they weren't enforcing any members in the subclasses and were requiring the effect type classes in  the methods themselves.
```scala
trait Blockstore[F[_]] {
  def put(…): F[Unit]
  def get(…): F[Option[BlockMessage]]
  …
}

class BlockStoreImpl[F[_]] extends BlockStore[F] {
  override def get(…)(implicit applicative: Applicative[F]) = …
}
```

but it's up to you of course.

Reply:
I agree that you should not have an implicit applicative member.